### PR TITLE
Setup users

### PR DESCRIPTION
**Pull request by Oluwadamilareolusakin**
# Description
Add user model(with devise authentications) and add post model(belongs to a user)

# What it does
It creates a user model with the corresponding database entries, it also adds devise functionality for user creations, authentications, password recovery, account confirmation. It adds a post model as well with the corresponding db table.

# Tests
* User and Post Model tests
* User and Post Feature tests

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class UsersController < ApplicationController
```
Comment (mosaaleb) [inline app/controllers/users_controller.rb]: - [ ] UsersController#index displays all users. Let's add it

**Comment by azdrenymeri** [comment]
### Hello @Oluwadamilareolusakin 👋 
#### Nice job on this milestone 🥇 
* There is this small issue:
[Screenshot](https://pasteboard.co/IBehyst.png)
* Also I wasn't able to create any posts
#### Kindly fix these issues and add it for a re-review

**Comment by Oluwadamilareolusakin** [comment]
@azdrenymeri hey man, so I just tested the posts creation and it works. May I know what challenges you faced please?

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,18 @@
 Rails.application.routes.draw do
+  root 'static_pages#timeline'
+  get '/dashboard', to: 'static_pages#dashboard'
+  
+  resources :posts, only: %i[ show destroy create ]
+
```
Comment (SunnyAsar) [inline config/routes.rb]: Hi, add routes for the user  as well.

Comment (Oluwadamilareolusakin) [reply config/routes.rb]: @sunnyasar thanks a lot for your kind words.

**Pull request merged**